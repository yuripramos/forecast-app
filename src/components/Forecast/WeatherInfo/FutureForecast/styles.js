import styled, { css } from "styled-components";
import { rem, media } from "../../../../styles/tools";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;
export const InnerContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(250)};
  display: inline-flex;
`;


export const Icon = styled.div`
  width: 100%;
  height: ${rem(60)};
`;

export const DaySpecs = styled.div`
  margin: ${rem(5)} ${rem(13)} ${rem(10)} ${rem(13)};
`;

export const Temperature = styled.h1`
  font-size: 1.5em;
  font-weight: 200;
  line-height: 0.75;
  font-family: "RobotoSlab Thin";
  margin-top: ${rem(20)};
  width: 100%;
   ${({ tiny }) =>
    tiny &&
    css`
      font-size: 1em;
  `};
`;

export const TinySpecs = styled.h3`
  color: white;
  font-weight: 100;
  line-height: 1.1;
  letter-spacing: 0.025em;
  margin: 0;
  padding: 0;
  font-family: "OpenSans Thin";
  width: 100%;
  margin: ${rem(10)} 0;
  :nth-child(2n) {
    margin: ${rem(2)} 0;
  }
`;
