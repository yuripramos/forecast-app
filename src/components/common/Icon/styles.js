import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools/index";

export const Span = styled.span`
  display: inline-flex;

  ${({ color }) =>
    color &&
    css`
      color: ${color};
    `};

  ${({ width, height }) =>
    width &&
    height &&
    css`
      svg {
        width: ${rem(width)};
        height: ${rem(height)};
      }
    `};
`;
