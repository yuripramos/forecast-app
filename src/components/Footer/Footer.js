import React from "react";
import { Links, ListLinks, ContentWrapper } from "./styles";
import { Container, Row, Column } from "../../styles/grid";
import { Menu } from "../../utils/getMenuFooter";

function Footer() {
  return (
    <ContentWrapper>
      <Container>
        <Row />
        <Row>
          <Column>
            <ListLinks>
              {Menu.map((c, i) => (
                <Links key={i}>{c}</Links>
              ))}
            </ListLinks>
          </Column>
        </Row>
      </Container>
    </ContentWrapper>
  );
}

export default Footer;
