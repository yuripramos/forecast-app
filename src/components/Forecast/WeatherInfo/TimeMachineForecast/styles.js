import styled, { css } from "styled-components";
import { rem, media } from "../../../../styles/tools";

export const InnerWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(250)};
  display: inline-flex;
  flex-wrap: wrap;
`;
