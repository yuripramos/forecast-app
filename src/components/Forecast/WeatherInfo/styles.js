import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
`;

export const City = styled.div`
  text-align: left;
  border-bottom: solid thin white;
  text-transform: uppercase;
  color: #e8e8e8;
  color: rgba(255, 255, 255, 0.7);
  font-family: "RobotoSlab Regular";
`;

export const Temperature = styled.h1`
  font-size: 4.5em;
  font-weight: 300;
  line-height: 0.75;
  font-family: "RobotoSlab Regular";
`;

export const TinySpecs = styled.h3`
  color: white;
  font-weight: 100;
  line-height: 1.1;
  letter-spacing: 0.025em;
  margin: 0;
  padding: 0;
  font-family: "RobotoSlab Regular";
`;

export const MainContentWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-content: flex-start;
  width: 70%;
  border: solid 1px blue;
`;

export const SidebarContentWrapper = styled.div`
  display: flex;
  justify-content: content;
  align-content: flex-end;
  width: 30%;
  border: solid 1px red;
`;
