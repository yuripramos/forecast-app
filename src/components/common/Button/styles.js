import React from "react";
import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { Link as RouterLink } from "react-router-dom";
import { lightBlue, white, green, green20 } from "../../../styles/settings";

export const ButtonTag = styled.button`
  background: ${lightBlue};
  font-family: "Helveticaneue Medium";
  border-radius: ${rem(4)};
  border: 0;
  cursor: pointer;
  display: inline-block;
  font-size: ${rem(12)};
  letter-spacing: ${rem(0.2)};
  line-height: ${rem(45)};
  outline: none;
  padding: 0 ${rem(20)};
  position: relative;
  color: ${white};
  text-align: center;
  text-decoration: none;
  transition: 0.2s ease-in-out color, 0.2s ease-in-out background,
    0.2s ease-in-out box-shadow, 0.2s ease-in-out border, 0.1s transform;
  white-space: nowrap;
  width: ${props => (props.width ? rem(props.width) : "100%")};
  user-select: none;
  opacity: ${props => (props.opacity && props.disabled ? props.opacity : 1)};

  ${media.md(css`
    width: ${props => (props.width ? rem(props.width) : "auto")};
  `)};

  ${media.lg(css`
    font-size: ${rem(13)};
    padding: 0 ${rem(28)};
    line-height: ${rem(48)};
  `)};

  :active {
    transform: translateY(${rem(1)});
  }

  ${({ isCallToAction }) =>
    isCallToAction &&
    css`
      background-color: ${green};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(100, 100, 100, 0.3);
      color: ${white};
      :hover {
        background: ${green20};
      }
    `};
`;

export const RouterLinkTag = ButtonTag.withComponent(
  ({ isCallToAction, ...props }) => <RouterLink {...props} /> // eslint-disable-line
);
