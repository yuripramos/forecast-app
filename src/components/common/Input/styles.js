import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools";
import { white, grey30, black50, darkBlue } from "../../../styles/settings";

export const InputField = styled.input`
  width: 30%;
  height: ${rem(25)};
  border-radius: ${rem(4)};
  background-color: ${white};
  border: solid ${rem(1)} ${grey30};
  letter-spacing: ${rem(0.8)};
  color: ${black50};
  padding: 0 ${rem(52)};
  transition: 0.3s padding, 0.3s border;

  ::-ms-clear {
    display: none !important;
  }

  :focus {
    outline: none;
    border: solid ${rem(1)} ${darkBlue};
  }
`;

export const InputWrapper = styled.div`
  width: 100%;
  display: inline-flex;
  position: relative;
  height: ${rem(26)};

  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};
`;

export const Label = styled.label`
  pointer-events: none;
  transition: 0.3s top, 0.2s color, 0.3s font-size;
  font-family: "Helveticaneue Medium";
  font-size: ${rem(16)};
  transform: translateY(5px);
`;
