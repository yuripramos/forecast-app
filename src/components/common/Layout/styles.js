import styled from "styled-components";
import { rem } from "../../../styles/tools";

export const Content = styled.div`
  min-height: calc(100vh - ${rem(500)});
`;
