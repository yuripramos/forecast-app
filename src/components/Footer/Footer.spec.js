import React from "react";
import { shallow, mount } from "enzyme";
import { Menu } from "../../utils/getMenuFooter";
import Footer from "./Footer";
import { Links } from "./styles";

describe("Footer component", () => {
  it("should match snapshot", () => {
    expect(shallow(<Footer />)).toMatchSnapshot();
  });
  it("should render menu items", () => {
    expect(mount(<Footer />).find(Links).length).toBe(Menu.length);
  });
});
