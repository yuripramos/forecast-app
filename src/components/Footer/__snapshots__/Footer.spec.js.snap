// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should match snapshot 1`] = `
<styled.div>
  <Container>
    <Row>
      <SocialNetworks
        description="San varius placerat. Vestibulum sed placerat ante, et varicerat ante, et varicerat ante, et"
        title="Say Hi & Get in Touch"
      />
    </Row>
    <Row>
      <Column>
        <styled.div>
          <styled.a
            key="0"
          >
            Contact
          </styled.a>
          <styled.a
            key="1"
          >
            Download
          </styled.a>
          <styled.a
            key="2"
          >
            Press
          </styled.a>
          <styled.a
            key="3"
          >
            Email
          </styled.a>
          <styled.a
            key="4"
          >
            Support
          </styled.a>
        </styled.div>
      </Column>
    </Row>
  </Container>
</styled.div>
`;
